### PR TITLE
Fix ivf getvector concurrence bug

### PR DESCRIPTION
**Pull request by foxspy**
issue : #132 
Fix the concurrence problem by moving the idmap construction into `build` stage.

### CODE REVIEW

File: src/index/ivf/ivf.cc [hunk 1]
```diff
@@ -317,13 +320,15 @@ IvfIndexNode<T>::Train(const DataSet& dataset, const Config& cfg) {
...
         }
         if constexpr (std::is_same<faiss::IndexBinaryIVF, T>::value) {
             const IvfBinConfig& ivf_bin_cfg = static_cast<const IvfBinConfig&>(cfg);
             auto nlist = MatchNlist(rows, ivf_bin_cfg.nlist.value());
             qzr = new (std::nothrow) typename QuantizerT<T>::type(dim, metric.value());
             index = std::make_unique<faiss::IndexBinaryIVF>(qzr, dim, nlist, metric.value());
             index->train(rows, (const uint8_t*)data);
+            index->make_direct_map(true);
         }
         index->own_fields = true;
```
Comment (cydrain) [inline src/index/ivf/ivf.cc:333]: you can set "index->make_direct_map(true)" here

Comment (foxspy) [reply src/index/ivf/ivf.cc:333]: Scann has not `make_direct_map` interface

**Comment by liliu-z** [comment]
Milvus issue: https://github.com/milvus-io/milvus/issues/27390

**Pull request merged**